cli/src/commands/codegen.ts

```diff
@@ -79,3 +114,73 @@ const deleteOldGeneratedFiles = (outDir: string) => {
     fs.unlinkSync(deletePath);
   }
 };
+
+const generateTypings = async (
+  outDir: string,
+  language: string,
+  appId: string,
+  shouldUseStaging: boolean,
+) => {
+  const manifests = await downloadEnabledPluginsManifests(
+    appId,
+    shouldUseStaging,
+  );
+  fs.writeFileSync(
+    path.join(outDir, "manifests.json"),
+    JSON.stringify(manifests, null, 2),
+  );
+  console.log("manifests.json generated successfully.");
+
+  if (language === "ts") {
+    // Aggregating all the manifests into a single file
+    let aggregatedManifest = `
+      syntax = 'proto3';
+      \n
+    `;
+
+    // Saving it to a temporary file
+    const osTmpDir = os.tmpdir();
+    const tmpFilePath = path.join(osTmpDir, "plugins.asterai.proto");
+    for (const manifest of manifests.manifests) {
+      aggregatedManifest += `${manifest.proto}\n`;
+    }
+    fs.writeFileSync(tmpFilePath, aggregatedManifest);
+
+    execSync(`
+      npx -p protobufjs-cli pbjs -t static --no-service ${tmpFilePath} -o ${path.join(outDir, "plugins.asterai.js")}
+    `);
+    execSync(`
+      npx -p protobufjs-cli pbts -o ${path.join(outDir, "plugins.asterai.d.ts")} ${path.join(outDir, "plugins.asterai.js")}
+    `);
+
+    fs.unlinkSync(tmpFilePath);
+    console.log("Typings generated successfully.");
+  }
+};
+
+const downloadEnabledPluginsManifests = async (
+  appId: string,
+  shouldUseStaging: boolean,
+): Promise<ExportedManifestFile> => {
+  const baseUrl = shouldUseStaging
+    ? STAGING_ENDPOINT_BASE_URL
+    : PRODUCTION_ENDPOINT_BASE_URL;
+  const response = await axios({
+    url: `${baseUrl}/app/${appId}/plugin/manifests`,
+    method: "GET",
+    headers: {
+      Authorization: getConfigValue("key"),
+    },
+  });
+
+  return response.data;
+};
+
+type ExportedManifestFile = {
```

Comment:
Response is probably a better name than 'file' since it includes multiple manifest files?
```suggestion
type ExportedManifestResponse = {
```